app/players/elves/elf.py

```diff
@@ -0,0 +1,17 @@
+from abc import ABC, abstractmethod
+
+from app.players.player import Player
+
+
+class Elf(Player, ABC):
+    @abstractmethod
+    def __init__(self,
+                 nickname: str,
+                 musical_instrument: str) -> None:
+        super().__init__(nickname)
+        self._musical_instrument = musical_instrument
+
+    @abstractmethod
+    def play_elf_song(self) -> None:
+        print(f"{self.nickname} is playing a song on the "
+              f"{self._musical_instrument}")
```

Comment:
This method contains the common logic for all elves playing a song. It should be a concrete method, not an abstract one. By marking it with @abstractmethod, you force every subclass of Elf to provide its own implementation, leading to duplicated code. If you remove the decorator, subclasses will inherit this method automatically.